sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -14247,4 +14250,24 @@ public void testLatestByOnStringColumnWithoutMaxBytesSpecified()
             new Object[] {"abc", defaultString, "def", defaultString, "def", defaultString}
         ));
   }
+
+  @Test
+  public void testUnSupportedNullsFirst()
+  {
+    DruidException e = assertThrows(DruidException.class, () -> testBuilder()
+        .queryContext(ImmutableMap.of(PlannerContext.CTX_ENABLE_WINDOW_FNS, true))
+        .sql("SELECT dim1,ROW_NUMBER() OVER (ORDER BY dim1 DESC NULLS FIRST) from druid.foo")
+        .run());
+    assertEquals("DESCENDING ordering with NULLS FIRST is not supported!", e.getMessage());
```

Comment:
There is a `DruidExceptionMatcher` which you can use to build matchers that test things like the persona and other parts of the exception which are also important.  On this test class, there are static helper methods `invalidSqlIs()` and `invalidSqlContains()` which will build a matcher checking the message.  Please use those instead.